### ISSUE TITLE
pyarmor obfuscation

### ISSUE THREAD

**Issue by JKKSI** [dashingsoft/pyarmor#498]
Hello there, have some question 

1 in a % 1 to 100 how much is possible to decrypt my python file with this command ?>>: pyarmor obfuscate --platform windows.x86 --platform linux.x86_64 --with-license licenses/r001/license.lic --plugin check_ntp_time --advanced 1 file.py 

2.  when i obfuscate my script with "--platform windows.x86 --platform linux.x86" , only works in linux, but in windows dont want work, why ?

3.- if i obfuscate with --advanced 4 , i can execute my script normally but will be more slow i guess? , what should be the best obfuscation command for my script for make including for professional people in cryptography really very very hard to decrypt my file?

4.- is there a command for make that people dont share the script to unlimited people untill the license expire?.

i ask because i want sell my script with licence "pyarmor licenses --expired 2050-05-20 r001" , but i'm some nervious, i hope your answer soon

**Comment by JKKSI** [comment]
other question i have is
when i obfuscate with --advanced 2 generate me this...
18/04/2021  11:12 p. m.    <DIR>          .
18/04/2021  11:12 p. m.    <DIR>          ..
18/04/2021  11:12 p. m.           552,004 Lightning_Destroyer.py
18/04/2021  11:12 p. m.         1,136,672 pytransform.cpython-38-x86_64-linux-gnu.so
18/04/2021  11:12 p. m.         1,224,718 pytransform.pyd
               3 archivos      2,913,394 bytes
               2 dirs  146,024,480,768 bytes libres

i'm in windows now, but "pytransform.cpython-38-x86_64-linux-gnu.so" i can delete this ?, or how i send the script obfuscated?, what happend if them delete pytransform.cpython-38-x86_64-linux-gnu.so?, pytransform.pyd is the carpet encrypted i guess?

**Comment by jondy** [comment]
> Hello there, have some question
> 
> 1 in a % 1 to 100 how much is possible to decrypt my python file with this command ?>>: pyarmor obfuscate --platform windows.x86 --platform linux.x86_64 --with-license licenses/r001/license.lic --plugin check_ntp_time --advanced 1 file.py
> 

I don't know how to answer this. Anyway refer to the security of pyarmor in pyarmor documentation

> 1. when i obfuscate my script with "--platform windows.x86 --platform linux.x86" , only works in linux, but in windows dont want work, why ?

If there is an issue, please following the issue template to report. I have no idea only by this information.

> 3.- if i obfuscate with --advanced 4 , i can execute my script normally but will be more slow i guess? , what should be the best obfuscation command for my script for make including for professional people in cryptography really very very hard to decrypt my file?
> 

The best solution is `--advanced 2`. `--advanced 4` is more hard to decrypt the script, but it's slow.

> 4.- is there a command for make that people dont share the script to unlimited people untill the license expire?.
> 
> i ask because i want sell my script with licence "pyarmor licenses --expired 2050-05-20 r001" , but i'm some nervious, i hope your answer soon

Please refer to how to use plugin to check internet time in the advanced usage of pyarmor documentation.

> other question i have is
> when i obfuscate with --advanced 2 generate me this...
> 18/04/2021 11:12 p. m.
> 
> .
> 18/04/2021 11:12 p. m. ..
> 18/04/2021 11:12 p. m. 552,004 Lightning_Destroyer.py
> 18/04/2021 11:12 p. m. 1,136,672 pytransform.cpython-38-x86_64-linux-gnu.so
> 18/04/2021 11:12 p. m. 1,224,718 pytransform.pyd
> 3 archivos 2,913,394 bytes
> 2 dirs 146,024,480,768 bytes libres
> i'm in windows now, but "pytransform.cpython-38-x86_64-linux-gnu.so" i can delete this ?, or how i send the script obfuscated?, what happend if them delete pytransform.cpython-38-x86_64-linux-gnu.so?, pytransform.pyd is the carpet encrypted i guess?

Just try it and get the result in 5 minutes.

**Comment by JKKSI** [comment]
> > other question i have is
> > when i obfuscate with --advanced 2 generate me this...
> > 18/04/2021 11:12 p. m.
> > .
> > 18/04/2021 11:12 p. m. ..
> > 18/04/2021 11:12 p. m. 552,004 Lightning_Destroyer.py
> > 18/04/2021 11:12 p. m. 1,136,672 pytransform.cpython-38-x86_64-linux-gnu.so
> > 18/04/2021 11:12 p. m. 1,224,718 pytransform.pyd
> > 3 archivos 2,913,394 bytes
> > 2 dirs 146,024,480,768 bytes libres
> > i'm in windows now, but "pytransform.cpython-38-x86_64-linux-gnu.so" i can delete this ?, or how i send the script obfuscated?, what happend if them delete pytransform.cpython-38-x86_64-linux-gnu.so?, pytransform.pyd is the carpet encrypted i guess?
> 
> Just try it and get the result in 5 minutes.

what do you mean with try it ? ,  i mean in my question, i obfuscate my script with --advanced 2, but generate me pytransform.cpython-38-x86_64-linux-gnu.so too, what happend if i delete this and i let only pytransform.pyd?, or i send the script with pytransform.cpython-38-x86_64-linux-gnu.so too ?

**Comment by JKKSI** [comment]
when i try to execute my script on linux this happends ...

┌──(root💀kali)-[/home/kali/Desktop/Lightning_Destroyer]
└─# ls                                                                                                                                                                                                                                 2 ⚙
Lightning_Destroyer.py  pytransform.cpython-38-x86_64-linux-gnu.so  pytransform.pyd
                                                                                                                                                                                                                                           
┌──(root💀kali)-[/home/kali/Desktop/Lightning_Destroyer]
└─# python3 Lightning_Destroyer.py                                                                                                                                                                                                     2 ⚙
Traceback (most recent call last):
  File "/home/kali/Desktop/Lightning_Destroyer/Lightning_Destroyer.py", line 2, in <module>
    from pytransform import pyarmor
ModuleNotFoundError: No module named 'pytransform'
                                                                                                                                                                                                                                           
┌──(root💀kali)-[/home/kali/Desktop/Lightning_Destroyer]
└─# chmod +x Lightning_Destroyer.py                                                                                                                                                                                                1 ⨯ 2 ⚙
                                                                                                                                                                                                                                           
┌──(root💀kali)-[/home/kali/Desktop/Lightning_Destroyer]
└─# python3 Lightning_Destroyer.py                                                                                                                                                                                                     2 ⚙
Traceback (most recent call last):
  File "/home/kali/Desktop/Lightning_Destroyer/Lightning_Destroyer.py", line 2, in <module>
    from pytransform import pyarmor
ModuleNotFoundError: No module named 'pytransform'
                                                                                                                                                                                                                                           
┌──(root💀kali)-[/home/kali/Desktop/Lightning_Destroyer]

**Comment by jondy** [comment]
Refer to 
https://pyarmor.readthedocs.io/en/latest/questions.html#python-import-system

**Comment by jondy** [comment]
> > > other question i have is
> > > when i obfuscate with --advanced 2 generate me this...
> > > 18/04/2021 11:12 p. m.
> > > .
> > > 18/04/2021 11:12 p. m. ..
> > > 18/04/2021 11:12 p. m. 552,004 Lightning_Destroyer.py
> > > 18/04/2021 11:12 p. m. 1,136,672 pytransform.cpython-38-x86_64-linux-gnu.so
> > > 18/04/2021 11:12 p. m. 1,224,718 pytransform.pyd
> > > 3 archivos 2,913,394 bytes
> > > 2 dirs 146,024,480,768 bytes libres
> > > i'm in windows now, but "pytransform.cpython-38-x86_64-linux-gnu.so" i can delete this ?, or how i send the script obfuscated?, what happend if them delete pytransform.cpython-38-x86_64-linux-gnu.so?, pytransform.pyd is the carpet encrypted i guess?
> > 
> > 
> > Just try it and get the result in 5 minutes.
> 
> what do you mean with try it ? , i mean in my question, i obfuscate my script with --advanced 2, but generate me pytransform.cpython-38-x86_64-linux-gnu.so too, what happend if i delete this and i let only pytransform.pyd?, or i send the script with pytransform.cpython-38-x86_64-linux-gnu.so too ?

Refer to https://pyarmor.readthedocs.io/en/latest/understand-obfuscated-scripts.html#runtime-package

**Comment by JKKSI** [comment]
> Refer to
> https://pyarmor.readthedocs.io/en/latest/questions.html#python-import-system

hello jondy, I apologize for giving so much trouble hehe, i checked the link that you gave me, but i still dont understand the problem, can you give me a example code or command line for solve this?, sorry and i hope your answer soon, thanks hehe

**Comment by JKKSI** [comment]
in windows the script execute normally without the error of import pytransform, but in linux gave me the error of import pytransform

**Comment by jondy** [comment]
What's your python version in Linux? If it's not python 3.8, it will not work.

**Comment by JKKSI** [comment]
> What's your python version in Linux? If it's not python 3.8, it will not work.

──(kali㉿kali)-[~]
└─$ python3 --version
Python 3.9.2
                                                                                                                                                                                                                                           
┌──(kali㉿kali)-[~]
└─$

**Comment by jondy** [comment]
The obfsucated scripts are bind to Python major/minor version. For example, if it’s obfuscated by Python 3.6, it must run by Python 3.6. It doesn’t work for Python 3.5

Please obfuscate the scripts by python 3.9 for linux, refer to
https://pyarmor.readthedocs.io/en/latest/advanced.html#obfuscating-scripts-by-other-python-version

**Comment by JKKSI** [comment]
> The obfsucated scripts are bind to Python major/minor version. For example, if it’s obfuscated by Python 3.6, it must run by Python 3.6. It doesn’t work for Python 3.5
> 
> Please obfuscate the scripts by python 3.9 for linux, refer to
> https://pyarmor.readthedocs.io/en/latest/advanced.html#obfuscating-scripts-by-other-python-version

hello jondy, i checked, and i obfuscated the script in windows but with python 3.8.3 , in my linux have └─$ python3 --version
Python 3.9.2
, i guess for this reason the script dont is working, i will try to obfuscate my script with python 3.9.2 and check if is working, if i get any bug or problem i will tell you :)

**Comment by JKKSI** [comment]
Hello there, i updated my python to the last version on my windows and linux too

in windows i have this next problem....

C:\Users\Daishinkan\Desktop\Beta\pyarmortryone\laboratorio>pyarmor -d obfuscate --platform windows.x86 --platform linux.x86_64 --with-license licenses/r001/license.lic --plugin check_ntp_time --advanced 2 Lightning_Destroyer.py
DEBUG    PyArmor installation path: C:\Users\Daishinkan\AppData\Roaming\Python\Python38\site-packages\pyarmor
DEBUG    PyArmor home path: C:\Users\Daishinkan\.pyarmor
DEBUG    Native platform is windows.x86
DEBUG    Search dynamic library in the path: C:\Users\Daishinkan\AppData\Roaming\Python\Python38\site-packages\pyarmor\platforms
DEBUG    Loaded dynamic library: C:\Users\Daishinkan\AppData\Roaming\Python\Python38\site-packages\pyarmor\platforms\windows\x86\_pytransform.dll
DEBUG    The version of core library is (41, 15, 7)
INFO     PyArmor Version 6.6.2
INFO     Target platforms: ['windows.x86', 'linux.x86_64']
DEBUG    Load platform list from C:\Users\Daishinkan\AppData\Roaming\Python\Python38\site-packages\pyarmor\platforms\index.json
DEBUG    Load platform list from C:\Users\Daishinkan\AppData\Roaming\Python\Python38\site-packages\pyarmor\platforms\index.json
INFO     Update target platforms to: ['windows.x86.11.py38', 'linux.x86_64.11.py38']
INFO     Source path is "C:\Users\Daishinkan\Desktop\Beta\pyarmortryone\laboratorio"
INFO     Entry scripts are ['Lightning_Destroyer.py']
INFO     Use cached capsule C:\Users\Daishinkan\.pyarmor\.pyarmor_capsule.zip
INFO     Search scripts mode: Normal
INFO     Save obfuscated scripts to "dist"
INFO     Read product key from capsule
INFO     Obfuscate module mode is 2
INFO     Obfuscate code mode is 1
INFO     Wrap mode is 1
INFO     Restrict mode is 1
INFO     Advanced value is 2
INFO     Super mode is True
INFO     Generating super runtime library to "dist"
DEBUG    Load platform list from C:\Users\Daishinkan\AppData\Roaming\Python\Python38\site-packages\pyarmor\platforms\index.json
DEBUG    Target dynamic library: ['C:\\Users\\Daishinkan\\.pyarmor\\platforms\\windows\\x86\\11\\py38\\pytransform.pyd', 'C:\\Users\\Daishinkan\\.pyarmor\\platforms\\linux\\x86_64\\11\\py38\\pytransform.cpython-38-x86_64-linux-gnu.so']
INFO     Extract pytransform.key
INFO     Generate license file from  licenses\r001\license.lic
INFO     Copying C:\Users\Daishinkan\.pyarmor\platforms\windows\x86\11\py38\pytransform.pyd
INFO     Patch extension dist\pytransform.pyd
DEBUG    Patching dist\pytransform.pyd
DEBUG    Found pattern at a42c4
DEBUG    Patch 261 bytes from a42e4
DEBUG    Patch 270 bytes from a43e9
DEBUG    Patch 420 bytes from a44f7
INFO     Patch library file OK
INFO     Copying C:\Users\Daishinkan\.pyarmor\platforms\linux\x86_64\11\py38\pytransform.cpython-38-x86_64-linux-gnu.so
INFO     Patch extension dist\pytransform.cpython-38-x86_64-linux-gnu.so
DEBUG    Patching dist\pytransform.cpython-38-x86_64-linux-gnu.so
DEBUG    Found pattern at b47c4
DEBUG    Patch 261 bytes from b47e4
DEBUG    Patch 270 bytes from b48e9
DEBUG    Patch 420 bytes from b49f7
INFO     Patch library file OK
INFO     Generate runtime files OK
INFO     Use protection template: C:\Users\Daishinkan\AppData\Roaming\Python\Python38\site-packages\pyarmor\protect_code2.pt
INFO     Start obfuscating the scripts...
INFO            Lightning_Destroyer.py -> dist\Lightning_Destroyer.py
INFO     Found plugin check_ntp_time at: C:\Users\Daishinkan\AppData\Roaming\Python\Python38\site-packages\pyarmor\plugins\check_ntp_time.py
INFO     Write patched script for debugging: C:\Users\Daishinkan\Desktop\Beta\pyarmortryone\laboratorio\Lightning_Destroyer.py.pyarmor-patched
INFO     Obfuscate 1 scripts OK.

C:\Users\Daishinkan\Desktop\Beta\pyarmortryone\laboratorio>cd dist

C:\Users\Daishinkan\Desktop\Beta\pyarmortryone\laboratorio\dist>py Lightning_Destroyer.py
Traceback (most recent call last):
  File "C:\Users\Daishinkan\Desktop\Beta\pyarmortryone\laboratorio\dist\Lightning_Destroyer.py", line 1, in <module>
    from pytransform import pyarmor
ImportError: DLL load failed while importing pytransform: %1 no es una aplicación Win32 válida.

C:\Users\Daishinkan\Desktop\Beta\pyarmortryone\laboratorio\dist>

in linux i have this problem ....
┌──(root💀kali)-[/home/kali/Desktop/Lightning_Destroyer]
└─# unzip dist.zip               
Archive:  dist.zip
  inflating: pytransform.pyd         
  inflating: Lightning_Destroyer.py  
  inflating: pytransform.cpython-38-x86_64-linux-gnu.so  
                                                                                                                                                                                                                                           
┌──(root💀kali)-[/home/kali/Desktop/Lightning_Destroyer]
└─# ls
dist.zip  Lightning_Destroyer.py  pytransform.cpython-38-x86_64-linux-gnu.so  pytransform.pyd
                                                                                                                                                                                                                                           
┌──(root💀kali)-[/home/kali/Desktop/Lightning_Destroyer]
└─# python3 Lightning_Destroyer.py 
Traceback (most recent call last):
  File "/home/kali/Desktop/Lightning_Destroyer/Lightning_Destroyer.py", line 1, in <module>
    from pytransform import pyarmor
ModuleNotFoundError: No module named 'pytransform'

can you help me providing me a command or something simillar for fix this?, i hope your answer soon hehe

**Comment by jondy** [comment]
In your windows, do the following commands work?

First create empty folder `test`, and create script `foo.py` with only one line `print('Hello')`, then
```
cd test
pyarmor obfuscate --advanced 2 foo.py
cd dist
python foo.py
```

**Comment by JKKSI** [comment]
> In your windows, check the following commands work or not?
> 
> First create empty folder `test`, and create script `foo.py` with only one line `print('Hello')`, then
> 
> ```
> cd test
> pyarmor obfuscate --advanced 2 foo.py
> cd dist
> python foo.py
> ```


C:\Users\Daishinkan\Desktop\Beta\pyarmortryone\laboratorio\test>pyarmor obfuscate --advanced 2 foo.py
INFO     PyArmor Version 6.6.2
INFO     Target platforms: Native
INFO     Update target platforms to: ['windows.x86.11.py38']
INFO     Source path is "C:\Users\Daishinkan\Desktop\Beta\pyarmortryone\laboratorio\test"
INFO     Entry scripts are ['foo.py']
INFO     Use cached capsule C:\Users\Daishinkan\.pyarmor\.pyarmor_capsule.zip
INFO     Search scripts mode: Normal
INFO     Save obfuscated scripts to "dist"
INFO     Read product key from capsule
INFO     Obfuscate module mode is 2
INFO     Obfuscate code mode is 1
INFO     Wrap mode is 1
INFO     Restrict mode is 1
INFO     Advanced value is 2
INFO     Super mode is True
INFO     Generating super runtime library to "dist"
INFO     Extract pytransform.key
INFO     Read default license from capsule
INFO     Copying C:\Users\Daishinkan\.pyarmor\platforms\windows\x86\11\py38\pytransform.pyd
INFO     Patch extension dist\pytransform.pyd
INFO     Patch library file OK
INFO     Generate runtime files OK
INFO     Use protection template: C:\Users\Daishinkan\AppData\Roaming\Python\Python38\site-packages\pyarmor\protect_code2.pt
INFO     Start obfuscating the scripts...
INFO            foo.py -> dist\foo.py
INFO     Obfuscate 1 scripts OK.

C:\Users\Daishinkan\Desktop\Beta\pyarmortryone\laboratorio\test>cd dist

C:\Users\Daishinkan\Desktop\Beta\pyarmortryone\laboratorio\test\dist>python foo.py
Traceback (most recent call last):
  File "foo.py", line 1, in <module>
    from pytransform import pyarmor
ImportError: DLL load failed: %1 no es una aplicaci¾n Win32 vßlida.

C:\Users\Daishinkan\Desktop\Beta\pyarmortryone\laboratorio\test\dist>

**Comment by jondy** [comment]
It seems this extension `pytransform.pyd` is not for this platform. 

Try to delete the whole folder `C:\Users\Daishinkan.pyarmor\platforms`, then obfuscate it again.

If it still doesn't work, could you check the dependencies of `test/dist/pytransform.pyd` by `dumpbin` or something else, for example
```
cd test/dist
dumpbin /dependents pytrasnform.pyd
```
Refer to #489

**Comment by JKKSI** [comment]
> It seems this extension `pytransform.pyd` is not for this platform.
> 
> Try to delete the whole folder `C:\Users\Daishinkan.pyarmor\platforms`, then obfuscate it again.
> 
> If it still doesn't work, could you check the dependencies of `test/dist/pytransform.pyd` by `dumpbin` or something else, for example
> 
> ```
> cd test/dist
> dumpbin /dependents pytrasnform.pyd
> ```
> 
> Refer to #489

i deleted all the folder that you give me and i try to obfuscate again, but still dont working, i try to check dependece of pytransform.pyd but i dont have dumpbin, i go to the "refer 489", but for be honest i dont know how to use the tool, i dont have dumpbin too, i check in youtube, but i see this is only for .exe and .dll, but i dont know where i search the dll

**Comment by JKKSI** [comment]
https://imgur.com/a/xCmjJDt , when i select all files in the rute that you give me test/dist/pytransform.pyd , and shows this https://imgur.com/a/xCmjJDt

**Comment by JKKSI** [comment]
my python versions on windows is...
C:\Users\Daishinkan\Desktop\Beta\pyarmortryone\laboratorio\test\dist>py --version
Python 3.9.4

C:\Users\Daishinkan\Desktop\Beta\pyarmortryone\laboratorio\test\dist>python --version
Python 2.7.12

C:\Users\Daishinkan\Desktop\Beta\pyarmortryone\laboratorio\test\dist>

**Comment by jondy** [comment]
Hi guy, you obfuscate the scripts by python 3.8, and run it by python 3.9 or python 2.7

Please make sure the python version used to obfuscate the scripts is same as the one to run the obfuscated scripts.

I don't support the basic usage of pyarmor once more.

**Comment by JKKSI** [comment]
> Hi guy, you obfuscate the scripts by python 3.8, and run it by python 3.9 or python 2.7
> 
> Please make sure the python version used to obfuscate the scripts is same as the one to run the obfuscated scripts.
> 
> I don't support the basic usage of pyarmor once more.

Hello Jondy, I Uninstalled succefully python 2.7.14 from my machine (windows 10), and i saw i had python 3.8.3 (i didnt saw it before sorry), i uninstalled him too succefully, later i reinstalled python 3.9.4, and this happened..... i learn the cmd say is no the path, but i dont know what would be the correctly path or how to fix this for try obfuscate again the script


C:\Users\Daishinkan>python --version
Python 3.9.4

C:\Users\Daishinkan>pip install pyarmor
WARNING: Value for scheme.headers does not match. Please report this to <https://github.com/pypa/pip/issues/9617>
distutils: c:\program files\python39\Include\UNKNOWN
sysconfig: c:\program files\python39\Include
WARNING: Additional context:
user = False
home = None
root = None
prefix = None
Defaulting to user installation because normal site-packages is not writeable
Collecting pyarmor
  Using cached pyarmor-6.6.2-py2.py3-none-any.whl (2.3 MB)
Installing collected packages: pyarmor
  WARNING: The script pyarmor.exe is installed in 'C:\Users\Daishinkan\AppData\Roaming\Python\Python39\Scripts' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
WARNING: Value for scheme.headers does not match. Please report this to <https://github.com/pypa/pip/issues/9617>
distutils: C:\Users\Daishinkan\AppData\Roaming\Python\Python39\Include\UNKNOWN
sysconfig: C:\Users\Daishinkan\AppData\Roaming\Python\Python39\Include
WARNING: Additional context:
user = True
home = None
root = None
prefix = None
Successfully installed pyarmor-6.6.2

C:\Users\Daishinkan>pip --version
pip 21.1 from C:\Users\Daishinkan\AppData\Roaming\Python\Python39\site-packages\pip (python 3.9)

C:\Users\Daishinkan>pyarmor -help
"pyarmor" no se reconoce como un comando interno o externo,
programa o archivo por lotes ejecutable.

C:\Users\Daishinkan>pyarmor obfuscate
"pyarmor" no se reconoce como un comando interno o externo,
programa o archivo por lotes ejecutable.

C:\Users\Daishinkan>

**Comment by JKKSI** [comment]
have same issue with python 3.9.2, and 3.8.6 i let you screenshots here https://imgur.com/a/GSk9t2M

**Comment by JKKSI** [comment]
hello jondy, later of working hard is working now pyarmor on windows .....

C:\Users\Daishinkan\Desktop\Beta\pyarmortryone\laboratorio\test>pyarmor obfuscate --advanced 2 foo.py
INFO     PyArmor Version 6.6.2
INFO     Target platforms: Native
INFO     Update target platforms to: ['windows.x86_64.11.py39']
INFO     Source path is "C:\Users\Daishinkan\Desktop\Beta\pyarmortryone\laboratorio\test"
INFO     Entry scripts are ['foo.py']
INFO     Use cached capsule C:\Users\Daishinkan\.pyarmor\.pyarmor_capsule.zip
INFO     Search scripts mode: Normal
INFO     Save obfuscated scripts to "dist"
INFO     Read product key from capsule
INFO     Obfuscate module mode is 2
INFO     Obfuscate code mode is 1
INFO     Wrap mode is 1
INFO     Restrict mode is 1
INFO     Advanced value is 2
INFO     Super mode is True
INFO     Generating super runtime library to "dist"
INFO     Create cross platforms path: C:\Users\Daishinkan\.pyarmor\platforms
INFO     Search library for platform: windows.x86_64.11.py39
INFO     Found available libraries: ['windows.x86_64.11.py39']
INFO     Target path for windows.x86_64.11.py39: C:\Users\Daishinkan\.pyarmor\platforms\windows\x86_64\11\py39
INFO     Downloading library file for windows.x86_64.11.py39 ...
INFO     Getting remote file: https://github.com/dashingsoft/pyarmor-core/raw/r41.15/platforms/windows.x86_64.11.py39/pytransform.pyd
INFO     Writing target file: C:\Users\Daishinkan\.pyarmor\platforms\windows\x86_64\11\py39\pytransform.pyd
INFO     Download dynamic library windows.x86_64.11.py39 OK
INFO     Extract pytransform.key
INFO     Read default license from capsule
INFO     Copying C:\Users\Daishinkan\.pyarmor\platforms\windows\x86_64\11\py39\pytransform.pyd
INFO     Patch extension dist\pytransform.pyd
INFO     Patch library file OK
INFO     Generate runtime files OK
INFO     Use protection template: c:\users\daishinkan\appdata\local\programs\python\python39\lib\site-packages\pyarmor\protect_code2.pt
INFO     Start obfuscating the scripts...
INFO            foo.py -> dist\foo.py
INFO     Obfuscate 1 scripts OK.

C:\Users\Daishinkan\Desktop\Beta\pyarmortryone\laboratorio\test>cd dist

C:\Users\Daishinkan\Desktop\Beta\pyarmortryone\laboratorio\test\dist>python foo.py
Hello

C:\Users\Daishinkan\Desktop\Beta\pyarmortryone\laboratorio\test\dist>dir
 El volumen de la unidad C no tiene etiqueta.
 El número de serie del volumen es: 3481-7480

 Directorio de C:\Users\Daishinkan\Desktop\Beta\pyarmortryone\laboratorio\test\dist

24/04/2021  12:10 p. m.    <DIR>          .
24/04/2021  12:10 p. m.    <DIR>          ..
24/04/2021  12:10 p. m.               976 foo.py
24/04/2021  12:10 p. m.         1,075,712 pytransform.pyd
               2 archivos      1,076,688 bytes
               2 dirs  142,705,360,896 bytes libres

C:\Users\Daishinkan\Desktop\Beta\pyarmortryone\laboratorio\test\dist>

**Comment by JKKSI** [comment]
in linux is working too 👯 👍 , thanks for have me patience hehe, and sorry for give much trouble :'v, but finally is working, i just needed update my python to the last version, restart pc and all working now.

**Comment by jondy** [comment]
There is a bug for super mode in windows.x86 with python 3.9, please refer to #489 to fix it at this time, it will be fixed in next version (maybe 6.7.0)

**Comment by JKKSI** [comment]
> There is a bug for super mode in windows.x86 with python 3.9, please refer to #489 to fix it at this time, it will be fixed in next version (maybe 6.7.0)

seems --advanced 1 give error too, i guess ;-;

C:\Users\Daishinkan\Desktop\ddos>pyarmor obfuscate --with-license licenses/r001/license.lic --plugin check_ntp_time --advanced 1 DDoS.py
INFO     PyArmor Version 6.6.2
WARNING  "--advanced 1" is deprecated for Python 3.9, use "--advanced 2" instead
INFO     Target platforms: Native
INFO     Source path is "C:\Users\Daishinkan\Desktop\ddos"
INFO     Entry scripts are ['DDoS.py']
INFO     Use cached capsule C:\Users\Daishinkan\.pyarmor\.pyarmor_capsule.zip
INFO     Search scripts mode: Normal
INFO     Save obfuscated scripts to "dist"
INFO     Read product key from capsule
INFO     Obfuscate module mode is 2
INFO     Obfuscate code mode is 1
INFO     Wrap mode is 1
INFO     Restrict mode is 1
INFO     Advanced value is 1
INFO     Super mode is False
INFO     Generating runtime files to dist\pytransform
INFO     Extract pytransform.key
INFO     Generate license file from  licenses\r001\license.lic
INFO     Copying c:\users\daishinkan\appdata\local\programs\python\python39\lib\site-packages\pyarmor\platforms\windows\x86_64\_pytransform.dll
INFO     Patch library dist\pytransform\_pytransform.dll
INFO     Patch library file OK
INFO     Copying c:\users\daishinkan\appdata\local\programs\python\python39\lib\site-packages\pyarmor\pytransform.py
INFO     Rename it to pytransform/__init__.py
INFO     Generate runtime files OK
INFO     Start obfuscating the scripts...
INFO            DDoS.py -> dist\DDoS.py
INFO     Found plugin check_ntp_time at: c:\users\daishinkan\appdata\local\programs\python\python39\lib\site-packages\pyarmor\plugins\check_ntp_time.py
INFO     Insert bootstrap code to entry script dist\DDoS.py
INFO     Obfuscate 1 scripts OK.

C:\Users\Daishinkan\Desktop\ddos>cd dist

C:\Users\Daishinkan\Desktop\ddos\dist>python DDoS.py
exception: access violation reading 0x0000000000000000

**Comment by jondy** [comment]
Do you really read issue #489?
